fetch.bs

```diff
@@ -7784,9 +7803,21 @@ method steps are:
   <ol>
    <li><p>If <var>locallyAborted</var> is true, terminate these substeps.
 
-   <li><p>If <var>response</var>'s <a for=response>aborted flag</a> is set, then <a>abort fetch</a>
-   with <var>p</var>, <var>request</var>, and <var>responseObject</var>, and terminate these
-   substeps.
+   <li><p>If <var>response</var>'s <a for=response>aborted flag</a> is set, then:
```

Comment:
In the actual fetch algorithm we also have this check and there is this line:

> If stream is [readable](https://streams.spec.whatwg.org/#readablestream-readable), [error](https://streams.spec.whatwg.org/#readablestream-error) stream with an "[AbortError](https://webidl.spec.whatwg.org/#aborterror)" [DOMException](https://webidl.spec.whatwg.org/#idl-DOMException).

I suspect that needs to be updated as well to use the abort reason on the controller?

It seems that the flow of this should be something like:

1. Website invokes `fetch()` with a signal.
2. Website aborts signal.
3. Signal is used in the fetch algorithm to terminate the actual fetch and cancel streams and such using the abort reason.
4. `fetch()` API observes the fetch algorithm being terminated and exposes the abort reason.

More concretely, why don't we use the abort reason on the controller in the steps below? I don't think we should end up in a situation where a response is aborted but the controller isn't yet.

Reply:
Good point, I've updated the erroring stream bit.

With regards to the latter part, I guess the reason why we don't use the abort reason on the controller here is because we need to abort the controller first which we do in Step 12.2.4 which in turn sets the serialized abort reason on the controller. Do you mean more that the controller does not need to be aborted since it should already be aborted at that point?

Reply:
Yeah, once the aborted flag is set the controller should also be aborted. I hope those cannot get out-of-sync... cc @noamr

Reply:
I see, in that case, I have updated the steps to use the controller's abort reason now. Hope it looks better!